system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/dataspaceconnector/test/e2e/EndToEndTransferTest.java

```diff
@@ -45,6 +45,7 @@
 import static org.awaitility.Awaitility.await;
 import static org.eclipse.dataspaceconnector.common.testfixtures.TestUtils.getFreePort;
 import static org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcessStates.COMPLETED;
+import static org.eclipse.dataspaceconnector.transfer.dataplane.spi.DataPlaneTransferType.SYNC;
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.CoreMatchers.notNullValue;
 
```

Comment:
This E2E test verifies the "HTTP pull" flow. I think we should write another `EndToEndTransferTest` for the "HTTP pull" flow using the `DataPlaneTransferFlowController` and `EmbeddedDataPlaneSelectorClient`.

Reply:
+1 but we can also add this in another PR

Reply:
I am ok with doing this in a separate PR as well. It still probably makes sense to do this work before starting #902 to prevent duplicating efforts with integration issues. WDYT @bscholtes1A ?

Reply:
There should be no duplication. Samples will rely on the embedded client, while the future e2e test will rely on separated runtimes for the Control Plane, DPF, and possibility the Selector. So for #902 can be started once the present PR is merged.

Reply:
After a discussion with the team we agreed we will start with a simpler sample with the embedded client (#902), and probably add a second one for the advanced setup with the remote DPF in a follow-up story. We can take over writing the e2e as part of this follow-up. Does this work for you @bscholtes1A ?